require.js

```diff
@@ -87,11 +83,20 @@
     if (typeof bundle === 'object') {
       for (var key in bundle) {
         if (has.call(bundle, key)) {
-          modules[key] = bundle[key];
+          require.register(modules[key], bundle[key]);
         }
       }
     } else {
       modules[bundle] = fn;
+      if (extExp.test(bundle)) {
+        var woExt = bundle.replace(extExp, '');
+        var existing = aliases[woExt];
+        if (!existing || existing !== woExt + '.js') aliases[woExt] = bundle;
+      }
+      if (indexExp.test(bundle)) {
+        var woIndex = bundle.replace(indexExp, '');
+        if (!has.call(aliases, woIndex)) aliases[woIndex] = bundle;
+      }
```

Comment:
This will also alias `p/a/t/h.coffee` to `p/a/t/h`, but if there are two files with the name whichever get's registered last will win. I shouldn't make an assumption here about `.js` being a preferred extension.

A few options:
 1. we can remove the `.js` preference and let either the first registered or last registered module to win, allowing the user to decide preference by order of inclusion. Templates could then be registered first or last, depending on which way we went.
 2. we could remove the extension alias completely (keeping only the index alias which is not new functionality) and let brunch set the aliases itself for extension or not.
 3. we could go more robust and allow registering extensions and using a priority to determine the winner.

3 is too much. I like 1 the best but 2 would be good too.